src/js/view/controls/tizen/tizen-controlbar.ts

```diff
@@ -0,0 +1,240 @@
+}
+
+export default class TizenControlbar extends Controlbar {
+    _api: PlayerAPI;
+    _model: ViewModel;
+    elements: any;
+    topLayout: ControlbarElement[];
+    bottomLayout: ControlbarElement[];
+    activeButton: Button | null;
+    el: HTMLElement;
+    nextUpToolTip: NextUpToolTip;
+    element: any;
+    on: any;
+    trigger: any;
+
+    constructor(_api: PlayerAPI, _model: ViewModel, _accessibilityContainer: any) {
+        super(_api, _model, _accessibilityContainer);
+
+        this._api = _api;
+        this._model = _model;
+        this.activeButton = null;
+        const localization = _model.get('localization');
+        const timeSlider = new TimeSlider(_model, _api, _accessibilityContainer.querySelector('.jw-time-update'));
+        
+        const superElements = this.elements;
+
+        const elements = this.elements = {
+            alt: superElements.alt,
+            play: superElements.play,
+            live: superElements.live,
+            elapsed: superElements.elapsed,
+            countdown: superElements.countdown,
+            time: timeSlider,
+            duration: superElements.duration,
+            buttonContainer: superElements.buttonContainer,
+            settingsButton: superElements.settingsButton,
+            back: button('jw-icon-back', () => {
+                this.trigger('backClick');
+            }, 'Back', cloneIcons('arrow-left')),
+            topContainer: div('jw-top-container'),
+            bottomContainer: div('jw-bottom-container')
+        };
+
+        // Remove play tooltip
+        elements.play.element().removeChild(elements.play.element().querySelector('.jw-tooltip-play'));
+
+        // Add button text tooltips
+        setTooltip(SimpleTooltip(elements.settingsButton.element(), 'settings', localization.settings));
+        setTooltip(SimpleTooltip(elements.back.element(), 'back', 'Back'));
+
+        const topLayout = this.topLayout = [
+            elements.back,
+            elements.settingsButton
+        ];
+
+        const bottomLayout = this.bottomLayout = [
+            elements.play,
+            elements.alt,
+            elements.live,
+            elements.elapsed,
+            elements.time,
+            elements.countdown
+        ];
+
+        const layout = [
+            elements.topContainer,
+            elements.buttonContainer,
+            elements.bottomContainer
+        ];
+
+        this.el = document.createElement('div');
+        this.el.className = 'jw-tizen-controlbar jw-controlbar jw-reset';
+
+        appendChildren(elements.topContainer, topLayout);
+        appendChildren(elements.bottomContainer, bottomLayout);
+        appendChildren(this.el, layout);
+
+        // Initial State
+        elements.play.show();
+        elements.back.show();
+    }
+
+    getRightButton(activeButton: Button, containerLayout: ControlbarElement[]): Button | undefined {
```

Comment:
i like this approach to dynamic button finding, i'm curious your thoughts on a way to potentially create an abstraction of `getRight` and `getLeft`. all of the logic is exactly the same except the for loop condition. it might be more trouble than it's worth but something to explore!

Reply:
+1 I 100% would love to do this. I spent a little time trying to figure out a solution, but will definitely explore some more.